data/core/units/monsters/Fungoid_Body.cfg

```diff
@@ -0,0 +1,275 @@
+#textdomain wesnoth-units
+
+[unit_type]
+    id=Fungoid Body
+    name= _ "Fungoid Body"
+    race=monster
+    image=units/monsters/core-compact.png
+    image_icon="units/monsters/core-compact.png~SCALE(72,72)"
+    [standing_anim]
+        [frame]
+            duration=100
+            image="units/monsters/core-compact.png"
+            auto_hflip=no
+        [/frame]
+    [/standing_anim]
+    [resistance]
+        blade=110
+        pierce=110
+        impact=80
+        fire=150
+        cold=150
+        arcane=200
+    [/resistance]
+    hitpoints=180
+    movement_type=smallfoot
+    movement=0
+    experience=150
+    {AMLA_DEFAULT}
+    level=3
+    alignment=chaotic
+    advances_to=null
+    cost=403
+    usage=mixed fighter
+    description= _ "This thing is impossible to describe, no one has seen anything like it before."
+    {DEFENSE_ANIM_FILTERED ("units/monsters/core-compact.png") ("units/monsters/core-compact.png") ("wail.wav") (direction=n,ne,se,s)}
+    {DEFENSE_ANIM_FILTERED ("units/monsters/core-compact.png~FL(horiz)") ("units/monsters/core-compact.png~FL(horiz)") ("wail.wav") (direction=nw,sw)}
+
+    [abilities]
+        {ABILITY_REGENERATES}
+        [dummy]
+            id=miasma
+            name= _ "miasma"
+            description=_ "This unit passively harms those not of its side at a distance and rate equal to its level."
+        [/dummy]
+    [/abilities]
+    # Miasma ability.
+    [event]
+        id=Fungoid Miasma
+        name=turn end
+        first_time_only=no
+        [store_unit]
+            [filter]
+                type="Fungoid Spore","Fungoid Stalk","Fungoid Body"
+            [/filter]
+            variable=fungoids
```

Comment:
If this miasma ability becomes mainline, filtering on ability instead of type would make it simpler to use more generally for UMC.

For this ability, it would be nice (necessary?) to have a visual indicator in the standing animation, some sort of halo cloud or something.  

If the ability is just an intended mechanic for the MP scenario you and Hejnewar are working on, then the filtering and standing animation probably don't need to be changed.

Reply:
> If this miasma ability becomes mainline, filtering on ability instead of type would make it simpler to use more generally for UMC.

Bah, I should have already thought of that, sorry.

> For this ability, it would be nice (necessary?) to have a visual indicator in the standing animation, some sort of halo cloud or something.

Yeah, that plus having the damage applied to all the victims at once would really speed up the presentation for the player while keeping it clear where the damage is coming from.